cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/ConstraintCheckerService.java

```diff
@@ -308,5 +322,25 @@ private ConstraintResult.SatisfiedState constraintsSatisfied(Job job, long now)
       return satisfiedState;
     }
 
+    private void emitScheduleJobSuccessAndLatencyMetric(Job job) {
+      if (metricsCollectionService == null) {
+        return;
+      }
+      MetricsContext collector = metricsCollectionService.getContext(METRICS_CONTEXT);
+      collector.increment("schedulejob.success", 1);
+
+      long currTime = System.currentTimeMillis();
+      long createdTime = job.getCreationTime();
+      long latency = currTime - createdTime;
+      collector.event("schedulejob.latency", latency);
+    }
+
+    private void emitScheduleJobFailureMetric() {
+      if (metricsCollectionService == null) {
+        return;
+      }
+      MetricsContext collector = metricsCollectionService.getContext(METRICS_CONTEXT);
+      collector.increment("schedulejob.failure", 1);
```

Comment:
This metric will be different from metric by the same name at https://github.com/cdapio/cdap/pull/14200/files#diff-468a931dd9015b0a2b89cb22f0b5ba4b578755418b25e4af437579a0e13679a3R109. Is that intentional? I wonder if it is helpful to have two different metrics for scheduling failures.

Reply:
This is intentional, both of the metrics will contribute to failure of a scheduled job. For differentiating the failures, we have set different context